internal/bminventory/inventory.go

```diff
@@ -2602,16 +2611,19 @@ func (b *bareMetalInventory) updateDhcpNetworkParams(updates map[string]interfac
 
 func (b *bareMetalInventory) setDiskEncryptionUsage(c *models.Cluster, diskEncryption *models.DiskEncryption, usages map[string]models.Usage) {
 
-	if c.DiskEncryption == nil {
+	if c.DiskEncryption == nil || reflect.DeepEqual(c.DiskEncryption, &models.DiskEncryption{}) {
 		return
 	}
 
-	props := map[string]interface{}{
-		"enable_on":    *diskEncryption.EnableOn,
-		"mode":         *diskEncryption.Mode,
-		"tang_servers": diskEncryption.TangServers,
+	props := map[string]interface{}{}
+	if diskEncryption.EnableOn != nil {
+		props["enable_on"] = swag.StringValue(diskEncryption.EnableOn)
+	}
+	if diskEncryption.Mode != nil {
+		props["mode"] = swag.StringValue(diskEncryption.Mode)
+		props["tang_servers"] = diskEncryption.TangServers
 	}
-	b.setUsage(*c.DiskEncryption.EnableOn != models.DiskEncryptionEnableOnNone, usage.DiskEncryption, &props, usages)
+	b.setUsage(swag.StringValue(c.DiskEncryption.EnableOn) != models.DiskEncryptionEnableOnNone, usage.DiskEncryption, &props, usages)
```

Comment:
I have tough about this case, there is no need for a check because `swag.StringValue()` returns `""` if the pointer is nil, therefore, if this is a `nil` value, the result will be `""` which mean that it is not `none` therefore the condition will be false.